src/coreclr/jit/lower.cpp

```diff
@@ -5753,6 +5753,36 @@ void Lowering::LowerShift(GenTreeOp* shift)
         shift->gtOp2->ClearContained();
     }
     ContainCheckShiftRotate(shift);
+
+#ifdef TARGET_ARM64
```

Comment:
It's unfortunate that we don't just create a "sbfiz/ubfiz" instruction node here instead of duplicating much of this same logic in codegen. Kind of one of the fallouts from not (generally) lowering to machine instructions.

Reply:
Is there something preventing us from doing that? We have some precedent here with `GT_BT` and some other opers.

Reply:
I don't think there's anything preventing us; it's a matter of cost an benefit.

A compiler I worked on in the past would, during "Lower", transform all machine-independent to machine-specific instructions, which all "lived" in the same namespace. RyuJIT decided not to do that, but to keep using machine-independent IR nodes all the way through codegen. Partially this is for (arguable) simplicity or (possibly) throughput, partially due to RyuJIT being an evolution. One-off machine-specific node types have been introduced.

Reply:
I don't remember why I didn't go this path, will check again